src/mutation.h

```diff
@@ -171,6 +182,22 @@ struct mutation_branch {
         /** The item, if any, spawned by the mutation */
         itype_id spawn_item;
 
+        /**Species ignoring character with the mutation*/
+        std::vector<species_id> ignored_by;
+
+        /**List of material required for food to be be edible*/
+        std::set<material_id> can_only_eat;
+
+        /**List of healing items allowed*/
+        std::vector<itype_id> can_only_heal_with;
+        std::vector<itype_id> can_heal_with;
+
+        /**List of allowed mutatrion category*/
+        std::vector<std::string> allowed_category;
```

Comment:
Those vectors should probably be `std::set`s. You don't care about the order, only about whether something is contained within them. That would also allow you to check this simpler:
```C++
if( can_only_heal_with.count( it.typeId() ) ) {
/* is usable */
} else {
/* not usable */
}
```